pkg/serverless/logs/http.go

```diff
@@ -32,10 +34,16 @@ func (l *LambdaLogsAPIServer) Close() {
 //
 //nolint:revive // TODO(SERV) Fix revive linter
 func (c *LambdaLogsAPIServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	data, _ := io.ReadAll(r.Body)
+	data, err := io.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("Error reading request body: %s", err)
+		w.WriteHeader(400)
+		return
+	}
 	defer r.Body.Close()
 	messages, err := parseLogsAPIPayload(data)
 	if err != nil {
+		log.Errorf("Error parsing Logs API payload: %s", err)
 		w.WriteHeader(400)
 	} else {
 		c.out <- messages
```

Comment:
If we can print the content of `messages` like [stream-logs](https://github.com/DataDog/datadog-agent/blob/33d43600c674a65c10336b0edadc2b677c6604a0/cmd/agent/subcommands/streamlogs/command.go#L56), it should give us a troubleshooting insight.